pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,195 @@
+// The current state of the operand is by watching the configs applied
+// to different static pod revisions that are active and uses the information to
+// update status as either progressing or completed or degraded.
+
+type LatencyProfileController struct {
+	operatorClient          v1helpers.StaticPodOperatorClient
+	targetNamespace         string
+	configClient            configv1.ConfigV1Interface
+	configMapClient         corev1client.ConfigMapsGetter
+	configNodeLister        listerv1.NodeLister
+	latencyProfileArgValMap map[apiconfigv1.WorkerLatencyProfileType][]nodeobserver.LatencyProfileArgVal
+}
+
+func NewLatencyProfileController(
+	operatorClient v1helpers.StaticPodOperatorClient,
+	targetNamespace string,
+	latencyProfileArgValMap map[apiconfigv1.WorkerLatencyProfileType][]nodeobserver.LatencyProfileArgVal,
+	configClient configv1.ConfigV1Interface,
+	nodeInformer configv1informers.NodeInformer,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	kubeClient kubernetes.Interface,
+	eventRecorder events.Recorder,
+) factory.Controller {
+
+	ret := &LatencyProfileController{
+		operatorClient:          operatorClient,
+		targetNamespace:         targetNamespace,
+		latencyProfileArgValMap: latencyProfileArgValMap,
+		configClient:            configClient,
+		configMapClient:         v1helpers.CachedConfigMapGetter(kubeClient.CoreV1(), kubeInformersForNamespaces),
+		configNodeLister:        nodeInformer.Lister(),
+	}
+
+	return factory.New().WithInformers(
+		// this is for our general configuration input and our status output in case another actor changes it
+		operatorClient.Informer(),
+
+		// We use nodeInformer for observing current worker latency profile
+		nodeInformer.Informer(),
+
+		// for configmaps of operator client target namespace
+		kubeInformersForNamespaces.InformersFor(targetNamespace).Core().V1().ConfigMaps().Informer(),
+	).ResyncEvery(time.Minute).WithSync(ret.sync).WithSyncDegradedOnError(operatorClient).ToController(
+		"WorkerLatencyProfile",
+		eventRecorder.WithComponentSuffix("latency-profile-controller"),
+	)
+}
+
+func (c *LatencyProfileController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+
+	// Collect the current latency profile
+	configNodeObj, err := c.configNodeLister.Get("cluster")
+
+	// if config/v1/node/cluster object is not found this controller should do nothing
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Warningf("nodes.config.openshift.io/cluster not found")
+			return nil
+		}
+		return err
+	}
+
+	// in case of empty workerlatency profile, set status Complete=False and pre-empt sync
+	if configNodeObj.Spec.WorkerLatencyProfile == "" {
+		// TODO: in case worker latency profile is transitioned
+		// from Default/Medium/Low to ""
+		// check that the required args should vanish
+		_, err = c.updateStatus(ctx, reasonLatencyProfileEmpty, "latency profile not set on cluster", false, true)
```

Comment:
are we `completed` when the profile wasn't set?

Reply:
reminder about the TODO

```
// TODO: in case worker latency profile is transitioned
// from Default/Medium/Low to ""
```

Reply:
It's taken care in the newest implementation anyway.